File: app/src/main/java/my/edu/tarc/jobseek/ui/headHunting/MyAdapter.java
package my.edu.tarc.jobseek.ui.headHunting;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import my.edu.tarc.jobseek.R;

public class MyAdapter extends RecyclerView.Adapter<MyViewHolder> {

    Context context;
    List<Candidate> candidates;

    public MyAdapter(Context context, List<Candidate> candidates) {
        this.context = context;
        candidates = candidates;
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new MyViewHolder(LayoutInflater.from(context).inflate(R.layout.recycler_view_row,parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {
        holder.textViewName.setText(candidates.get(position).getName());
        holder.textViewSkills.setText(candidates.get(position).getSkills().toString());
        holder.textViewName.setText(candidates.get(position).getPosition());
        holder.textViewName.setText(candidates.get(position).getLocation());

    }

    @Override
    public int getItemCount() {
        return candidates.size();
    }
}

File: app/src/main/java/my/edu/tarc/jobseek/headHunting/MyAdapter.kt
package my.edu.tarc.jobseek.headHunting

import android.content.Context
import android.content.DialogInterface
import android.content.DialogInterface.OnClickListener
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.ViewParent
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import my.edu.tarc.jobseek.R
import org.w3c.dom.DocumentType
import org.w3c.dom.Text

class MyAdapter(
    private val context: Context,
    private val recordClickListener: RecordClickListener
) :
    RecyclerView.Adapter<MyAdapter.CandiViewHolder>() {
    private var candList = emptyList<Candidate>()

    class CandiViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val name: TextView = itemView.findViewById(R.id.textViewCandiName)
        val field: TextView = itemView.findViewById(R.id.textViewCandiField)
        val experience: TextView = itemView.findViewById(R.id.textViewCandiExp)
        val location: TextView = itemView.findViewById(R.id.textViewCandiLocation)
    }

    internal fun setCandi(candiList: List<Candidate>) {
        this.candList = candiList
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CandiViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.recycler_view_layout, parent, false)
        return CandiViewHolder(view)
    }

    override fun onBindViewHolder(holder: CandiViewHolder, position: Int) {
        val candi = candList[position]
        holder.name.text = candi.name
        holder.field.text = candi.field
        holder.experience.text = candi.experience.toString()
        holder.location.text = candi.location
        holder.itemView.setOnClickListener {
        }


    }

    override fun getItemCount(): Int {
        return candList.size
    }
}

    // onClickListener Interface
    interface RecordClickListener {
        fun onRecordClickListener(index: Int)
    }
